clients/rust/src/lib.rs

```diff
@@ -71,16 +73,43 @@ impl SysadminClient {
     pub fn connect<A: ToSocketAddrs>(&mut self, address: A) -> SysadminResult<()> {
         let stream = TcpStream::connect(address).context("failed to connect to sysadmin")?;
         stream
-            .set_write_timeout(Some(self.timeout.clone()))
+            .set_write_timeout(Some(self.timeout))
             .context("error setting write timeout")?;
         stream
-            .set_read_timeout(Some(self.timeout.clone()))
+            .set_read_timeout(Some(self.timeout))
             .context("error setting read timeout")?;
         self.stream = Some(stream);
 
         Ok(())
     }
 
+    /// Set TCP stream timeout
+    pub fn set_timeout(&mut self, timeout: Duration) -> SysadminResult<()> {
+        if self.stream.is_none() {
+            bail!(SysadminErrorKind::SysadminConnectionError(
+                "Command issues before connection was initialized".to_owned()
+            ));
+        }
+
+        self.timeout = timeout;
+
+        if let Some(stream) = self.stream.as_mut() {
+            stream
+                .set_write_timeout(Some(self.timeout))
+                .context("Error setting write timeout")?;
+            stream
+                .set_read_timeout(Some(self.timeout))
+                .context("Error setting read timeout")?;
+        }
+
+        Ok(())
+    }
+
+    /// Reset TCP stream timeout to default (10 seconds).
+    pub fn default_timeout(&mut self) -> SysadminResult<()> {
```

Comment:
I don't like the naming of this function - seems like it would return a `Duration` the way it is now

Reply:
Yeah I see that, I think I'm going to go with your suggestion of renaming to something like `reset_timeout_to_default`. Unless I do a `default_timeout` field in `SysadminClient` & make this an getter